src/controller/tests/data_model/TestRead.cpp

```diff
@@ -295,6 +354,316 @@ void TestReadInteraction::TestReadDataVersionFilter(nlTestSuite * apSuite, void
+
+    MockInteractionModelApp delegate;
+    chip::app::AttributeCache cache(delegate);
+    auto * engine = chip::app::InteractionModelEngine::GetInstance();
+    err           = engine->Init(&ctx.GetExchangeManager());
+    NL_TEST_ASSERT(apSuite, err == CHIP_NO_ERROR);
+
+    chip::app::AttributePathParams attributePathParams1[3];
+    attributePathParams1[0].mEndpointId  = Test::kMockEndpoint2;
+    attributePathParams1[0].mClusterId   = Test::MockClusterId(3);
+    attributePathParams1[0].mAttributeId = Test::MockAttributeId(1);
+
+    attributePathParams1[1].mEndpointId  = Test::kMockEndpoint2;
+    attributePathParams1[1].mClusterId   = Test::MockClusterId(3);
+    attributePathParams1[1].mAttributeId = Test::MockAttributeId(2);
+
+    attributePathParams1[2].mEndpointId  = Test::kMockEndpoint3;
+    attributePathParams1[2].mClusterId   = Test::MockClusterId(2);
+    attributePathParams1[2].mAttributeId = Test::MockAttributeId(2);
+
+    chip::app::AttributePathParams attributePathParams2[2];
+    attributePathParams2[0].mEndpointId  = Test::kMockEndpoint2;
+    attributePathParams2[0].mClusterId   = Test::MockClusterId(3);
+    attributePathParams2[0].mAttributeId = kInvalidAttributeId;
+
+    attributePathParams2[1].mEndpointId  = Test::kMockEndpoint3;
+    attributePathParams2[1].mClusterId   = Test::MockClusterId(2);
+    attributePathParams2[1].mAttributeId = Test::MockAttributeId(2);
+
+    chip::app::AttributePathParams attributePathParams3[3];
+    attributePathParams3[0].mEndpointId  = Test::kMockEndpoint1;
+    attributePathParams3[0].mClusterId   = Test::MockClusterId(2);
+    attributePathParams3[0].mAttributeId = kInvalidAttributeId;
+
+    attributePathParams3[1].mEndpointId  = Test::kMockEndpoint2;
+    attributePathParams3[1].mClusterId   = Test::MockClusterId(3);
+    attributePathParams3[1].mAttributeId = kInvalidAttributeId;
+
+    attributePathParams3[2].mEndpointId  = Test::kMockEndpoint2;
+    attributePathParams3[2].mClusterId   = Test::MockClusterId(2);
+    attributePathParams3[2].mAttributeId = kInvalidAttributeId;
+
+    chip::app::EventPathParams eventPathParams[100];
+    for (uint32_t index = 0; index < 100; index++)
+    {
+        eventPathParams[index].mEndpointId = Test::kMockEndpoint3;
+        eventPathParams[index].mClusterId  = Test::MockClusterId(2);
+        eventPathParams[index].mEventId    = 0;
+    }
+
+    chip::app::ReadPrepareParams readPrepareParams(ctx.GetSessionBobToAlice());
+    readPrepareParams.mMinIntervalFloorSeconds   = 0;
+    readPrepareParams.mMaxIntervalCeilingSeconds = 4;
+    //
+    // Test the application callback as well to ensure we get the right number of SubscriptionEstablishment/Termination
+    // callbacks.
+    //
+    app::InteractionModelEngine::GetInstance()->RegisterReadHandlerAppCallback(&gTestReadInteraction);
+
+    int testId = 0;
+    // Initial Read of C1A1, C1A2 and C2A1. Cache has two C1 attributes at version 0, and one C2 attribute at version 0.
```

Comment:
If you're going to use terminology like C1A1, we should line up consts that you can use in the path params as well to match up the logic as well.